src/core/document.js

```diff
@@ -349,22 +349,23 @@ var PDFDocument = (function PDFDocumentClosure() {
   var EMPTY_FINGERPRINT = '\x00\x00\x00\x00\x00\x00\x00' +
     '\x00\x00\x00\x00\x00\x00\x00\x00\x00';
 
-  function PDFDocument(pdfManager, arg, password) {
+  function PDFDocument(pdfManager, arg, password, filename) {
```

Comment:
Remove `filename` from `PDFDocument`. It currently does not even know about the URL of the document, so it shouldn't know about the filename either.

PDFManager has the filename, so one can simply use `pdfDocument.pdfManager.filename` if wanted.